docs/structure-of-a-contract.rst

```diff
@@ -126,6 +126,40 @@ Events are convenience interfaces with the EVM logging facilities.
 See :ref:`events` in contracts section for information on how events are declared
 and can be used from within a dapp.
 
+.. _structure-errors:
+
+Errors
+======
+
+Errors allow you to define descriptive names and data for failure situations.
+Errors can be used in :ref:`revert statements <revert-statement>`.
+In comparison to string descriptions, errors are much cheaper and allow you
+to encode additional data. You can use NatSpec to describe the error to
+the user.
+
+::
+
+    // SPDX-License-Identifier: GPL-3.0
+    pragma solidity ^0.8.3;
+
+    /// Not enough funds for transfer. Requested `requested`,
+    /// but only `available` available.
+    error NotEnoughFunds(uint requested, uint available);
+
+    contract Token {
+        mapping(address => uint) balances;
+        function transfer(address to, uint amount) public {
+            uint balance = balances[msg.sender];
+            if (balance < amount)
+                revert NotEnoughFunds(amount, balance);
```

Comment:
If there is the same error `NotEnoughFunds` also defined within the contract, what is right now the behaviour?

Reply:
This is not possible, you would get an "identifier already defined" error.